ts/components/dialog/OnionStatusPathDialog.tsx

```diff
@@ -119,19 +136,17 @@ const OnionPathModalInner = () => {
           </StyledLightsContainer>
           <Flex container={true} flexDirection="column" alignItems="flex-start">
             {nodes.map((snode: Snode | any) => {
-              let labelText = snode.label
-                ? snode.label
-                : countryLookup.byIso(ip2country(snode.ip))?.country;
-              if (!labelText) {
-                labelText = window.i18n('unknownCountry');
-              }
-              return labelText ? (
+              const countryName =
+                reader?.get(snode.ip || '0.0.0.0')?.country?.names[lang] ||
+                window.i18n('unknownCountry');
```

Comment:
can this not be some sort of memorized function that takes in the snode ip as an arg?